src/DataIntegrity.php

```diff
@@ -105,6 +108,38 @@ public static function ensureFieldsPresent(dict<string, mixed> $row, table_schem
               } else {
                 $row[$field_name] = (int)$row[$field_name];
               }
+            } else if ($row[$field_name] is int) {
+
+                $row[$field_name] = (int)$row[$field_name];
```

Comment:
it shouldn't be necessary to cast here since we know it's an int. we could do `$row[$field_name] as int;` to convince the type checker, but possibly better would be to pull this into a local variable above the first if statement as `$field_value`, that way type refinement will kick in and hack will know it's an int.